src/bin/sol-fbp-generator/main.c

```diff
@@ -634,7 +634,15 @@ generate_node_type_assignments(const struct fbp_data *data)
 
     SOL_VECTOR_FOREACH_IDX (&data->graph.nodes, n, i) {
         struct type_description *desc = n->user_data;
-        out("    nodes[%d].type = %s;\n", i, desc->symbol);
+        struct sol_str_slice module;
+        char *sep;
+
+        module = sol_str_slice_from_str(desc->name);
+        sep = strstr(module.data, "/");
+        if (sep)
+            module.len = sep - module.data;
```

Comment:
On Fri, 18 Sep 2015 14:39:48 -0700, Leandro A. F. Pereira wrote:
> > @@ -634,7 +634,15 @@ generate_node_type_assignments(const struct fbp_data *data)
> >  
> >      SOL_VECTOR_FOREACH_IDX (&data->graph.nodes, n, i) {
> >          struct type_description *desc = n->user_data;
> > -        out("    nodes[%d].type = %s;\n", i, desc->symbol);
> > +        struct sol_str_slice module;
> > +        char *sep;
> > +
> > +        module = sol_str_slice_from_str(desc->name);
> > +        sep = strstr(module.data, "/");
> > +        if (sep)
> > +            module.len = sep - module.data;
> 
> Shouldn't module.data be updated as well?

No, why? It's already pointing at the beginning of the string as set by
sol_str_slice_from_str()